app/furniture/journal/keyword.rb

```diff
@@ -7,9 +7,11 @@ class Keyword < ApplicationRecord
     belongs_to :journal, inverse_of: :keywords
 
     def self.extract_and_create_from!(body)
-      body.scan(/#(\w+)/)&.flatten&.each do |keyword|
-        next if where(":aliases = ANY (aliases)", aliases: keyword)
-          .or(where(canonical_keyword: keyword)).exists?
+      body.scan(/#(\w+)/)&.flatten&.map do |keyword|
+        existing_keyword = where(":aliases = ANY (aliases)", aliases: keyword)
+          .or(where(canonical_keyword: keyword)).first
```

Comment:
I realize this query is from before your PR, but reading it over right now it struck me that it feels like it should be a nice little scope, which would help tighten up this entire loop a bit.

Reply:
Agree; I would love for this to be a scope!